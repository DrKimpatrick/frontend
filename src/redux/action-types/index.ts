// This is where we will add action types