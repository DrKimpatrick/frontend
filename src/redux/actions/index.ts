// This is where we will add our action creators
export {};
