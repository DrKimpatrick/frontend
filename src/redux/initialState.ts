import users from 'redux/initialStates/users';

const initialState = {
  users
};

export default initialState;
