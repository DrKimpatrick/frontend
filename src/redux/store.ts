import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import apiMiddleware from 'middleware/apiMiddleware';
import rootReducer from './reducers/rootReducer';
import initialState from './initialState';

const middleware = [apiMiddleware];

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export type RootState = ReturnType<typeof rootReducer>;

export default store;
