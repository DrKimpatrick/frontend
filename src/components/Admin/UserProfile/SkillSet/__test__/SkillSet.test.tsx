import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { Store } from 'redux';
import Component from '../SkillSet';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { cleanup, render } from '@testing-library/react';

let store: Store;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const userSkill = {
  skill: {
    _id: '5fb8cebd750b0131d5a18cc2',
    skill: 'Typescript',
    createdAt: '2020-11-21T08:24:29.802Z',
    updatedAt: '2020-11-21T08:24:29.802Z'
  },
  level: 'advanced',
  verificationStatus: 'unverified',
  _id: '5fb8d5cb750b0131d5a18cd0',
  createdAt: '2020-11-21T08:54:35.322Z',
  updatedAt: '2020-11-21T08:54:35.322Z'
};

const initialState = {
  users: {
    userSkillLoading: false,
    userSkill: [
      { ...userSkill },
      { ...userSkill, level: 'beginner' },
      { ...userSkill, level: 'intermediate' }
    ]
  }
};

const props = {
  userId: 'id'
};

describe('SkillSet', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={store}>
        <Component {...props} />
      </Provider>,
      div
    );
  });

  it('should create snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Component {...props} />
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });

  describe('Advanced skill', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        users: {
          ...initialState.users,
          userSkill: [
            { ...userSkill, level: 'beginner' },
            { ...userSkill, level: 'intermediate' }
          ]
        }
      });
    });
    it('should return message when advanced skill not found', () => {
      const { container } = render(
        <Provider store={store}>
          <Component {...props} />
        </Provider>
      );

      const message = container.querySelector('.notFound h5');

      expect(message).toHaveTextContent('There are no advanced skills');
    });
  });

  describe('Intermediate skill', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        users: {
          ...initialState.users,
          userSkill: [
            { ...userSkill, level: 'beginner' },
            { ...userSkill, level: 'advanced' }
          ]
        }
      });
    });
    it('should return message when intermediate skill not found', () => {
      const { container } = render(
        <Provider store={store}>
          <Component {...props} />
        </Provider>
      );

      const message = container.querySelector('.notFound h5');

      expect(message).toHaveTextContent('There are no intermediate skills');
    });
  });

  describe('Beginner skill', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        users: {
          ...initialState.users,
          userSkill: [
            { ...userSkill, level: 'intermediate' },
            { ...userSkill, level: 'advanced' }
          ]
        }
      });
    });
    it('should return message when beginners skill not found', () => {
      const { container } = render(
        <Provider store={store}>
          <Component {...props} />
        </Provider>
      );

      const message = container.querySelector('.notFound h5');

      expect(message).toHaveTextContent('There are no beginner skills');
    });
  });

  describe('loading spinner', () => {
    beforeEach(() => {
      store = mockStore({
        ...initialState,
        users: { ...initialState.users, userSkillLoading: true }
      });
    });

    it('should return loading spinner', () => {
      const { container } = render(
        <Provider store={store}>
          <Component {...props} />
        </Provider>
      );

      const loading = container.querySelector('.sideLoading');

      expect(loading).toBeDefined();
    });
  });
});
