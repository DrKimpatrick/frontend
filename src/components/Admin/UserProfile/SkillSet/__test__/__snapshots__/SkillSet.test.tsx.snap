// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillSet should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="items"
        >
          <div
            class="title"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
              />
            </svg>
            <h1>
              Skillset
            </h1>
          </div>
          <div
            class="skillDivider"
          >
            <div
              class="title"
            >
              <h5
                class="beginner"
              >
                Beginners
              </h5>
            </div>
            <div
              class="listItems"
            >
              <ul
                class="p-0 m-2 bg-card-preview"
              >
                <li
                  class="py-2 py-2 rounded-sm"
                >
                  <h5>
                    Typescript
                  </h5>
                  <div
                    class="selectItems"
                  >
                    <select>
                      <option
                        value="verified"
                      >
                        verified
                      </option>
                      <option
                        value="unverified"
                      >
                        unverified
                      </option>
                      <option
                        value="inProgress"
                      >
                        inProgress
                      </option>
                    </select>
                    <span
                      class="customArrow"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </span>
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="skillDivider"
          >
            <div
              class="title"
            >
              <h5
                class="intermediate"
              >
                Intermediate
              </h5>
            </div>
            <div
              class="listItems"
            >
              <ul
                class="p-0 m-2 bg-card-preview"
              >
                <li
                  class="py-2 py-2 rounded-sm"
                >
                  <h5>
                    Typescript
                  </h5>
                  <div
                    class="selectItems"
                  >
                    <select>
                      <option
                        value="verified"
                      >
                        verified
                      </option>
                      <option
                        value="unverified"
                      >
                        unverified
                      </option>
                      <option
                        value="inProgress"
                      >
                        inProgress
                      </option>
                    </select>
                    <span
                      class="customArrow"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </span>
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="skillDivider"
          >
            <div
              class="title"
            >
              <h5
                class="advanced"
              >
                Advanced
              </h5>
            </div>
            <div
              class="listItems"
            >
              <ul
                class="p-0 m-2 bg-card-preview"
              >
                <li
                  class="py-2 py-2 rounded-sm"
                >
                  <h5>
                    Typescript
                  </h5>
                  <div
                    class="selectItems"
                  >
                    <select>
                      <option
                        value="verified"
                      >
                        verified
                      </option>
                      <option
                        value="unverified"
                      >
                        unverified
                      </option>
                      <option
                        value="inProgress"
                      >
                        inProgress
                      </option>
                    </select>
                    <span
                      class="customArrow"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </span>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class="items"
      >
        <div
          class="title"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
            />
          </svg>
          <h1>
            Skillset
          </h1>
        </div>
        <div
          class="skillDivider"
        >
          <div
            class="title"
          >
            <h5
              class="beginner"
            >
              Beginners
            </h5>
          </div>
          <div
            class="listItems"
          >
            <ul
              class="p-0 m-2 bg-card-preview"
            >
              <li
                class="py-2 py-2 rounded-sm"
              >
                <h5>
                  Typescript
                </h5>
                <div
                  class="selectItems"
                >
                  <select>
                    <option
                      value="verified"
                    >
                      verified
                    </option>
                    <option
                      value="unverified"
                    >
                      unverified
                    </option>
                    <option
                      value="inProgress"
                    >
                      inProgress
                    </option>
                  </select>
                  <span
                    class="customArrow"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="skillDivider"
        >
          <div
            class="title"
          >
            <h5
              class="intermediate"
            >
              Intermediate
            </h5>
          </div>
          <div
            class="listItems"
          >
            <ul
              class="p-0 m-2 bg-card-preview"
            >
              <li
                class="py-2 py-2 rounded-sm"
              >
                <h5>
                  Typescript
                </h5>
                <div
                  class="selectItems"
                >
                  <select>
                    <option
                      value="verified"
                    >
                      verified
                    </option>
                    <option
                      value="unverified"
                    >
                      unverified
                    </option>
                    <option
                      value="inProgress"
                    >
                      inProgress
                    </option>
                  </select>
                  <span
                    class="customArrow"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="skillDivider"
        >
          <div
            class="title"
          >
            <h5
              class="advanced"
            >
              Advanced
            </h5>
          </div>
          <div
            class="listItems"
          >
            <ul
              class="p-0 m-2 bg-card-preview"
            >
              <li
                class="py-2 py-2 rounded-sm"
              >
                <h5>
                  Typescript
                </h5>
                <div
                  class="selectItems"
                >
                  <select>
                    <option
                      value="verified"
                    >
                      verified
                    </option>
                    <option
                      value="unverified"
                    >
                      unverified
                    </option>
                    <option
                      value="inProgress"
                    >
                      inProgress
                    </option>
                  </select>
                  <span
                    class="customArrow"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
